### PR TITLE
[EH] update credential type to AsyncTokenCredential

### PR DESCRIPTION
**Pull request by swathipil**
fixes: #17796

### CODE REVIEW

**Review by yunhaoling — APPROVED** [review]

File: sdk/eventhub/azure-eventhub/azure/eventhub/aio/_consumer_client_async.py [hunk 1]
```diff
@@ -136,7 +136,7 @@ def __init__(
         fully_qualified_namespace: str,
         eventhub_name: str,
         consumer_group: str,
-        credential: Union["TokenCredential", AzureSasCredential],
+        credential: Union["AsyncTokenCredential", AzureSasCredential],
```
Comment (yunhaoling) [inline sdk/eventhub/azure-eventhub/azure/eventhub/aio/_consumer_client_async.py:139]: docstring also needs change

File: sdk/eventhub/azure-eventhub/azure/eventhub/aio/_producer_client_async.py [hunk 2]
```diff
@@ -76,7 +76,7 @@ def __init__(
         self,
         fully_qualified_namespace: str,
         eventhub_name: str,
-        credential: Union["TokenCredential", AzureSasCredential],
+        credential: Union["AsyncTokenCredential", AzureSasCredential],
```
Comment (yunhaoling) [inline sdk/eventhub/azure-eventhub/azure/eventhub/aio/_producer_client_async.py:79]: docstring also needs change

File: sdk/eventhub/azure-eventhub/azure/eventhub/aio/_consumer_client_async.py (hunk 1 above)
Comment (swathipil) [reply sdk/eventhub/azure-eventhub/azure/eventhub/aio/_consumer_client_async.py:139]: thought I did this 🤔 weird, fixing

Comment (swathipil) [reply sdk/eventhub/azure-eventhub/azure/eventhub/aio/_consumer_client_async.py:139]: ah the merge must have undone it

**Pull request merged**